packages/react/src/tabs/list/TabsList.tsx

```diff
@@ -1,28 +1,118 @@
 'use client';
 import * as React from 'react';
 import { BaseUIComponentProps } from '../../utils/types';
-import { useComponentRenderer } from '../../utils/useComponentRenderer';
+import { useEventCallback } from '../../utils/useEventCallback';
+import { useModernLayoutEffect } from '../../utils/useModernLayoutEffect';
+import { useRenderElement } from '../../utils/useRenderElement';
 import { CompositeRoot } from '../../composite/root/CompositeRoot';
 import { tabsStyleHookMapping } from '../root/styleHooks';
 import { useTabsRootContext } from '../root/TabsRootContext';
-import { TabsRoot } from '../root/TabsRoot';
-import { type TabMetadata } from '../tab/useTabsTab';
-import { useTabsList } from './useTabsList';
+import { type TabsRoot, type TabsOrientation, type TabActivationDirection } from '../root/TabsRoot';
+import { type TabMetadata } from '../tab/TabsTab';
 import { TabsListContext } from './TabsListContext';
 
 const EMPTY_ARRAY: number[] = [];
 
+function getInset(tab: HTMLElement, tabsList: HTMLElement) {
```

Comment:
nit: could you move these helper functions below the component? This way, while reading the code we'll go from high to low level details.